astroid/__pkginfo__.py

```diff
@@ -15,7 +21,12 @@
 
 extras_require = {}
 install_requires = ['lazy_object_proxy', 'six', 'wrapt']
-extras_require[':python_version<"3.4"'] = ['enum34', 'singledispatch']
+
+if py_version < (3, 4) and setuptools_version < '21.0.0':
```

Comment:
setuptools 3.3 is sufficient for the environment marker "`python_version`", so you can use
`setuptools_version < (3, 3)`